### PR TITLE
Moving gRPC stub-with-auth helpers out of bigtable package.

### PR DESCRIPTION
**Pull request by dhermes**
Also changing the metadata plugin to take credentials and user agent directly instead of inferring those values from a client (this is because the Bigtable client is unique in it's attributes).

This is in preparation for gRPC support for datastore as is #2081

### CODE REVIEW

File: gcloud/_helpers.py [hunk 1]
```diff
@@ -522,6 +526,68 @@ def _name_from_project_path(path, project, template):
...
 
+class MetadataPlugin(object):
+    """Callable class to transform metadata for gRPC requests.
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials`
+    :param credentials: The OAuth2 Credentials to use for creating
+                        access tokens.
+
+    :type user_agent: str
+    :param user_agent: (Optional) The user agent to be used with API requests.
```
Comment (tseaver) [inline gcloud/_helpers.py]: `(Optional)` appears not to be true.

File: gcloud/_helpers.py [hunk 2]
```diff
@@ -522,6 +526,68 @@ def _name_from_project_path(path, project, template):
...
+    """
+
+    def __init__(self, credentials, user_agent):
+        self._credentials = credentials
+        self._user_agent = user_agent
+
+    def __call__(self, unused_context, callback):
+        """Adds authorization header to request metadata."""
```
Comment (tseaver) [inline gcloud/_helpers.py]: Parameter documentation?

**Comment by tseaver** [comment]
LGTM modulo docstring comments.

File: gcloud/_helpers.py (hunk 1 above)
Comment (dhermes) [inline gcloud/_helpers.py]: D'oh. Copy-pasta. Thanks.

**Pull request merged**